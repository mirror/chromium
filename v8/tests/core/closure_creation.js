// The following piece of code illustrates that we do not compile
// several versions of functions when they can be shared (have the
// same context).

var g0;
var g10;
var g20;

function f(x) {

  function g() {
    return 1;
  }

  switch (x) {
    case 0:
      g0 = g;
      return;
    case 10:
      g.foo = "foo10";
      g10 = g;
      break;
    case 20:
      g.foo = "foo20";
      g20 = g;
      break;
  }

  f(x - 1);
}


f(100);
print(g10());  // 1
print(g20());  // 1
if (g10 === g20) {
  print(g10.foo == "foo10" && g20.foo == "foo10");  // true
} else {
  print(g10.foo == "foo10" && g20.foo == "foo20");  // true
}
print(typeof g0);
print(typeof g10);
print(typeof g20);
