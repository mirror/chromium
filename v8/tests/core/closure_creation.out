1
1
true
function
function
function
